demos/CoreClrConsoleApplications/HelloWorld/HelloWorld.cs

```diff
@@ -301,4 +314,36 @@ @@@@@@@@@@@         @@@@@@@@@@@@@
         Console.ResetColor();
         Console.WriteLine();
     }
+
+    //HACK: Needed until CoreFX implements Environment.OSVersion
+    private static string DetectOS()
+    {
+        Process proc = new Process();
+
+        string detectionResult = String.Empty;
+
+        if (proc.StartInfo.Environment.ContainsKey("TERM_PROGRAM"))
+        {
+            detectionResult = "mac";
+        }
+        else if (proc.StartInfo.Environment.ContainsKey("SHELL"))
+        {
+            string shellValue = proc.StartInfo.Environment["SHELL"].ToLowerInvariant();
+
+            if (shellValue.EndsWith("/tcsh") || shellValue.EndsWith("/sh"))
```

Comment:
I wouldn't question an assertion that bash is the most popular shell on Linux, but I do question an assertion that no one on Linux uses tcsh or sh.  (Or bash executed as /bin/sh).

Really I think I'm just in @stephentoub's camp here: for best (consistency in) results copy the PlatformDetection logic from CoreFx (pending it being officially queryable API)